pkg/smart_wallet/factory_calldata.go

```diff
@@ -0,0 +1,70 @@
+package smart_wallet
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/shopspring/decimal"
+)
+
+func GetFactoryCallData(smartWalletConfig Config, ownerAddress common.Address, index decimal.Decimal) ([]byte, error) {
+	var initCode []byte
+	var err error
+
+	switch typ := *smartWalletConfig.Type; typ {
+	case SimpleAccountType:
+		return nil, fmt.Errorf("%w: %s", ErrSmartWalletNotSupported, typ)
+	case BiconomyType: // not tested
+		initCode, err = GetBiconomyFactoryCallData(ownerAddress, index, smartWalletConfig.ECDSAValidator)
+	case KernelType:
+		initCode, err = GetKernelFactoryCallData(ownerAddress, index, smartWalletConfig.Logic, smartWalletConfig.ECDSAValidator)
+	default:
+		return nil, fmt.Errorf("unknown smart wallet type: %s", typ)
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to get init code: %w", err)
+	}
+
+	return initCode, nil
+}
+
+// getKernelInitData returns the calldata needed call the factory
+// to deploy a Zerodev Kernel smart account.
+func GetKernelFactoryCallData(owner common.Address, index decimal.Decimal, accountLogic, ecdsaValidator common.Address) ([]byte, error) {
```

Comment:
What about making `GetKernelFactoryCallData` private again since it has an entry point function `GetFactoryCallData`?

I don't have context here but should the package allow its users to reference `GetKernelFactoryCallData` directly? If so, it should be mentioned in README as well.

Reply:
Re this see [this comment](https://github.com/layer-3/clearsync/pull/229#discussion_r1530328975)